website/content/docs/overview/architecture/index.md

```diff
@@ -9,25 +9,55 @@ weight: 10
+ and custom resource definition in order to define its work items and reconcile the state of
+ them.
+
+The primary interaction object for a data user is the `M4DApplication` CRD where a user defines which
+data should be used for which purpose. The following chart and description describe the architecture and components of
+{{< name >}} relative to when they are used.
+
+![Architecture](workflow_multicluster.png)
+
+Before the data user can perform any actions a data operator has to install the {{< name >}} and modules
+that can be used by the user. [Modules](../modules) are a way that the {{< name >}} abstracts data access for
+users. Modules can provide read/write access or produce implicit copies that serve as lower latency caches of
+remote assets. Modules also enforce policies defined for data assets. A detailed description of them can be found [here](../modules).
+
+The {{< name >}} connects to external services to retrieve information about policies, available data and credentials.
+Policies, assets and access credentials to the assets have to be defined before the user can create an application. The
+current abstraction knows 3 different connectors but depending on the implementation the underlying system
+could be the same (e.g. a data catalog could store asset definitions and credentials). More information about connectors
+can be found [here](../connectors). It's designed in an open way so that multiple different catalog and policy frameworks
+of all kind of cloud and on-prem systems can be supported. The data steward configures policies in an external policy manager
+over assets defined in an external data catalog.
+
+Once a developer submits a `M4DApplication` CRD to Kubernetes the ApplicationController will make sure that all
+the specs are fulfilled and will make sure that the data is made accessible to the user according to the
+previously defined policies. The `M4DApplication` holds metadata about the application such the data assets required by the 
+application, the processing purpose and the method of access the user wishes (protocol e.g. S3 or Arrow flight). 
+It uses this information to check with external systems (4) if access is allowed
+and whether restrictive policies such as masking or hashing have to be applied. It compiles blueprints depending
+on the policies defined by the connectors and chooses which modules (5) are best fit for the requirements that the user 
+specified regarding the access protocol and availability.
+As data assets may reside in different systems the 
+blueprints are compiled in a `Plotter` CRD (6) that specifies which blueprints have to be executed in which cluster.
+
+Depending on the setup the `PlotterController` will use various methods to distribute the blueprints. In a multi cluster
+setup [Razee](http://razee.io) is used for the control. The `PlotterController` will also collect statuses and distribute
```

Comment:
can you make Razee an example? maybe saying that M4D can support multiple multi-cloud tools, but only Razee is implemented so far.